src/auto/unix_input_stream.rs

```diff
@@ -0,0 +1,84 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use InputStream;
+use PollableInputStream;
+use gio_sys;
+use glib::object::Cast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect_raw;
+use glib::translate::*;
+use glib_sys;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::mem::transmute;
+
+glib_wrapper! {
+    pub struct UnixInputStream(Object<gio_sys::GUnixInputStream, gio_sys::GUnixInputStreamClass, UnixInputStreamClass>) @extends InputStream, @implements PollableInputStream;
+
+    match fn {
+        get_type => || gio_sys::g_unix_input_stream_get_type(),
+    }
+}
+
+impl UnixInputStream {
+    pub fn new(fd: i32, close_fd: bool) -> UnixInputStream {
+        unsafe {
+            InputStream::from_glib_full(gio_sys::g_unix_input_stream_new(fd, close_fd.to_glib())).unsafe_cast()
+        }
+    }
+}
+
+pub const NONE_UNIX_INPUT_STREAM: Option<&UnixInputStream> = None;
+
+pub trait UnixInputStreamExt: 'static {
+    fn get_close_fd(&self) -> bool;
+
+    fn get_fd(&self) -> i32;
```

Comment:
Via `FromRawFd` (and does not need to be unsafe). You probably also want to implement `AsRawFd` on both streams.

Reply:
To satisfy the FromRawFd docs, get_fd should probably be defined as moving self so that the UnixInputStream is no longer usable, right?

Reply:
The problem here is that all `glib::Object`s are reference counted. So even if you take self by value, another reference could still be around elsewhere.

For `FromRawFd`, I meant an implementation like the one for the socket: https://github.com/gtk-rs/gio/blob/047ea1cf0735f04c5a404331c61d192ac4cb43a9/src/socket.rs#L197-L202

Reply:
And note that this is (probably?) fine as every API doing something with raw fds must be unsafe (see above), so whenever dealing with raw fds it's the programmer's job to ensure that things are done correctly.

Reply:
Done